### PR TITLE
Add a system prediff for programs launched with OpenMPI's mpirun.

### PR DESCRIPTION
**Pull request by gbtitus**
The OpenMPI mpirun system launcher used by the mpirun4ofi Chapel
launcher produces helpful output messages when programs exit with
nonzero status, but these messages interfere with sub_test pass/fail
analysis for such tests, even when the nonzero status is expected.
Here, add a system prediff to filter out these mpirun messages.

### CODE REVIEW

**Comment by ronawho** [comment]
Would throwing `--quiet` to mpirun limit the amount filtering we need to do?

File: util/test/prediff-for-mpirun4ofi [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+outfileTmp = open(sys.argv[2]+".tmp" , "w")
+msg = outfile.read()
+msg = msg1aPat.sub("", msg)
+msg = msg1bPat.sub("", msg)
+msg = msg2Pat.sub("", msg)
+outfileTmp.write(msg)
+outfileTmp.close()
+outfile.close()
+
+subprocess.call("mv "+sys.argv[2]+".tmp"+" "+sys.argv[2], shell=True )
```
Comment (ronawho) [inline util/test/prediff-for-mpirun4ofi]: https://docs.python.org/3/library/os.html#os.rename

**Comment by gbtitus** [comment]
> Would throwing `--quiet` to mpirun limit the amount filtering we need to do?

It looks like it prevents the "mpirun detected that one or more processes exited with non-zero status ..." message from being printed.  Since that will simplify the filtering, I'll add `--quiet`.

**Comment by ronawho** [comment]
Not super important, because this is just a prediff, but personally I find this a little clearer still as something like:

```python
import sys, re

# These match the message blocks we want to remove.
msgs = (
"""
^-------------------------------------------------------
Primary job +terminated normally, but [0-9]+ process[^ ]* returned
a non-zero exit code[.]+ Per user-direction, the job has been aborted\.
-------------------------------------------------------
""",
"""
^-------------------------------------------------------
While the primary job +terminated normally, [0-9]+ process[^ ]* returned
a non-zero exit code[.]+ Further examination may be required\.
-------------------------------------------------------
"""
)

outfname = sys.argv[2]

with open(outfname, 'r') as f:
    output = f.read()

for msg in msgs:
    output = re.sub(msg, '', output, flags=re.MULTILINE)

with open(outfname, 'w') as f:
    f.write(output)
```

File: util/test/prediff-for-mpirun4ofi [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#
+# or messages like this:
+#     -------------------------------------------------------
+#     While the primary job  terminated normally, 1 process returned
+#     a non-zero exit code.. Further examination may be required.
+#     -------------------------------------------------------
+#
+# This script removes such messages.
+#
+import sys, os, re
```
Comment (ronawho) [inline util/test/prediff-for-mpirun4ofi]: Can remove os import now

Comment (gbtitus) [reply util/test/prediff-for-mpirun4ofi]: D'oh!  Crud, thought I'd nailed it.

**Pull request merged**